### ISSUE TITLE
Quick question about the get_rgbd_point_cloud function

### ISSUE THREAD

**Issue by lukemelas** [facebookresearch/co3d#41]
Dear authors,

Thank you for your great dataset and code. I am looking to transform an image and corresponding depth map into a point cloud. I see that you have helpfully provided a function `get_rgbd_point_cloud` in the point cloud utils file for this purpose. However, I am having difficulty with this function. 

Specifically, I am able to create a point cloud, but it does not have the expected shape. I've created a minimal example below:
```python
import torch

from pytorch3d.implicitron.dataset.dataloader_zoo import FrameData
from pytorch3d.implicitron.dataset.dataset_zoo import dataset_zoo
from pytorch3d.implicitron.tools.point_cloud_utils import get_rgbd_point_cloud
from pytorch3d.structures import Pointclouds
from pytorch3d.vis.plotly_vis import plot_scene


# Dataset arguments (copied from the single-sequence implicitron config)
dataset_args = {
    'dataset_name': 'co3d_singlesequence',
    'dataset_root': '/path/to/co3d',
    'category': 'hydrant',
    'limit_to': -1,
    'limit_sequences_to': -1,
    'n_frames_per_sequence': 1,
    'test_on_train': False,
    'load_point_clouds': True,
    'mask_images': False,
    'mask_depths': False,
    'restrict_sequence_name': (),
    'test_restrict_sequence_id': 0,
    'assert_single_seq': True,
    'only_test_set': False,
    'aux_dataset_kwargs': {
        'box_crop': True,
        'box_crop_context': 0.3,
        'image_width': 800,
        'image_height': 800,
        'remove_empty_masks': True
    },
    'path_manager': None
}
# Load dataset
datasets = dataset_zoo(**dataset_args)

# Get first item from the dataset
item: FrameData = datasets['train'][0]

# Check shapes
print(item.image_rgb.shape)  # -> torch.Size([3, 800, 800])
print(item.depth_map.shape)  # -> torch.Size([1, 800, 800])
print(item.depth_mask.shape)  # -> torch.Size([1, 800, 800])

# Get point cloud
rendered_pointcloud = get_rgbd_point_cloud(
    camera=item.camera,
    image_rgb=torch.unsqueeze(item.image_rgb, dim=0),
    depth_map=item.depth_map,
    mask=item.depth_mask,
    mask_thr=0.50,
)

# Check shapes
print(rendered_pointcloud.points_packed().shape)  # -> torch.Size([97710, 3])

# Plot a single point cloud using plotly
plot_scene({'Pointcloud': {'scene': rendered_pointcloud}})

# Get point cloud with constant depth
rendered_pointcloud = get_rgbd_point_cloud(
    camera=item.camera,
    image_rgb=torch.unsqueeze(item.image_rgb, dim=0),
    depth_map=item.depth_map,
    mask=item.depth_mask,
    mask_thr=0.50,
)

# Check shapes
print(rendered_pointcloud.points_packed().shape)  # -> torch.Size([97710, 3])

# Plot a single point cloud using plotly
plot_scene({'Pointcloud': {'scene': rendered_pointcloud}})

```

I obtain the following point cloud: 

![gif](https://s7.gifyu.com/images/ezgif-5-9473a7275a.gif)

(This is with a different sequence_id from the minimal example above, but the result is similar)

I was expecting something along the lines of the point cloud in `item.sequence_point_cloud` (although of course only a partial point cloud, because I only have a single view here). When I visualize that point cloud, it renders as expected.

I realize that I am most likely just misunderstanding something and using this function incorrectly, but I'm not sure how I should be using it.

Thank you and all the best,
Luke

**Comment by davnov134** [comment]
Hi,

`get_rgbd_point_cloud ` accepts a batch of image_rgb, depth_map and depth_mask.
So make sure to pass in tensors of shape `[B, dim, H, W]` (i.e. add a singleton dimension to your current input) and see whether this fixes your issues.

**Comment by lukemelas** [comment]
Thanks for the very quick response. I was adding a batch dimension to the image, but not the depth map, which led to a silent broadcasting error. Expanding the depth map resolved the issue. 

Closing this issue, thanks once again.
